source/WAVChunk.cpp

```diff
@@ -177,4 +177,28 @@ bool WAVChunk::IsEqual(WAVChunk &wavChunk)
 	bIsEqual = (bBaseEqual && bIsEqual);
 
 	return bIsEqual;
+}
+
+std::shared_ptr<nlohmann::json> WAVChunk::ToJSON()
+{
+	auto JSONDocument = nlohmann::json();
+	auto strChunkName = ChunkTypesNamingUtility::toString(GetChunkType());
+
+	// Adding in Basechunk fields
+	JSONDocument[strChunkName]["SourceIndentifierSize"] = std::to_string(m_u16SourceIndentifierSize);
+	JSONDocument[strChunkName]["SourceIndentifier"] = m_vu8SourceIdentifier;
+
+	// Adding in WAVChunk fields
+	JSONDocument[strChunkName]["WAVHeader"] = GetHeaderString(); // not sure if we want to nest JSON representation here rather?
+	JSONDocument[strChunkName]["ChunkSize"] = std::to_string(GetSize());
+	JSONDocument[strChunkName]["TimeStamp"] = std::to_string(m_i64TimeStamp);
+
+	std::stringstream stream;
+
+	for (size_t i = 0; i < m_vi16Data.size(); ++i) {
```

Comment:
Just some formatting suggestions:

- For the single line for lopps, remove the `{` and `}`
- Try not to use `i`, `j`, `k`, when using for loops. For simple loops it is fine but once one need loops in loops or uses iterators readability of the code drops quickly. I have generally used things like `szDatumIndex` or `uDatumIndex` when using loop indicies

Reply:
Done, I'm sure we can set up a linter to check these things